services/QuillLMS/engines/comprehension/app/controllers/comprehension/rules_controller.rb

```diff
@@ -45,7 +45,7 @@ def destroy
     end
 
     private def set_rule
-      @rule = Comprehension::Rule.find(params[:id])
+      @rule = Comprehension::Rule.find_by_uid(params[:id]) || Comprehension::Rule.find(params[:id])
```

Comment:
This doesn't affect Comprehension work I've worked on, but I can't speak to the entire app

Reply:
I consider `Comprehension::Rule.find_by_uid(params[:id])` to be an antipattern. It should be `params[:uid]`.  You'd have decompose the simplicity of the RESTful route to do this, however, so at minimum let's add a warning comment here